src/test/java/org.spongepowered.mod/test/integration/CreeperTest.java

```diff
@@ -0,0 +1,136 @@
+ * THE SOFTWARE.
+ */
+package org.spongepowered.mod.test.integration;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import com.flowpowered.math.vector.Vector3d;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.block.BlockSnapshot;
+import org.spongepowered.api.block.BlockTypes;
+import org.spongepowered.api.data.Transaction;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.type.HandTypes;
+import org.spongepowered.api.entity.EntityTypes;
+import org.spongepowered.api.entity.living.monster.Creeper;
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.entity.living.player.gamemode.GameModes;
+import org.spongepowered.api.event.EventListener;
+import org.spongepowered.api.event.block.ChangeBlockEvent;
+import org.spongepowered.api.event.cause.EventContextKeys;
+import org.spongepowered.api.event.entity.MoveEntityEvent;
+import org.spongepowered.api.event.entity.SpawnEntityEvent;
+import org.spongepowered.api.item.ItemTypes;
+import org.spongepowered.api.item.inventory.ItemStack;
+import org.spongepowered.api.item.inventory.entity.Hotbar;
+import org.spongepowered.api.item.inventory.query.QueryOperationTypes;
+import org.spongepowered.mctester.internal.BaseTest;
+import org.spongepowered.mctester.internal.event.StandaloneEventListener;
+import org.spongepowered.mctester.api.junit.MinecraftRunner;
+import org.spongepowered.mctester.junit.TestUtils;
+import org.spongepowered.mctester.api.WorldOptions;
+
+@RunWith(MinecraftRunner.class)
+@WorldOptions(deleteWorldOnSuccess = true)
+public class CreeperTest extends BaseTest {
+
+    public CreeperTest(TestUtils testUtils) {
+        super(testUtils);
+    }
+
+    @Test
+    public void explodeCreeper() throws Throwable {
+        this.testUtils.runOnMainThread(() -> {
+            Player player = this.testUtils.getThePlayer();
+            player.offer(Keys.GAME_MODE, GameModes.CREATIVE);
+
+            ((Hotbar) player.getInventory().query(QueryOperationTypes.INVENTORY_TYPE.of(Hotbar.class))).setSelectedSlotIndex(0);
+            player.setItemInHand(HandTypes.MAIN_HAND, ItemStack.builder().itemType(ItemTypes.SPAWN_EGG).quantity(1).add(Keys.SPAWNABLE_ENTITY_TYPE, EntityTypes.CREEPER).build());
+        });
+
+        testUtils.waitForInventoryPropagation();
+
+        // Look at the ground two blocks in the z direction
+        Vector3d targetPos = this.testUtils.runOnMainThread(() -> CreeperTest.this.testUtils.getThePlayer().getLocation().getPosition().add(0, -1, 2));
+        client.lookAt(targetPos);
+
+        final Creeper[] creeper = new Creeper[1];
```

Comment:
Eh, this way of capturing a creeper instance looks weird (though I know you cannot use non-final local variables in lambda expressions. Any way to make it look better?)

Reply:
I just went with Intellij's suggestion - I don't know of a nicer way.